examples/fit_ODNP_data.py

```diff
@@ -0,0 +1,126 @@
+"""Fitting ODNP Datasets for Ksigma
+===================================
+The T1(p) and E(p) integrals are stored in
+previous post processing pulled from an H5 file.
+The inverse of the T1(p) data processing is taken
+to produce the R1(p) data which is fit using the
+polyfit of the inverse of krho with two degrees of
+freedom. The R1(p) fit is then fed into the
+fitting routine for the enhancement data prior to
+normalization. The cross relaxivity is then
+calculated using the output of these fits as well
+as the sample parameters that are fed to in (e.g.,
+ppt value, and concentration).
+"""
+from pyspecdata import *
+from sympy import symbols, Symbol, latex,lambdify
+from scipy.io import loadmat
+
+h5_file = "ras.h5"
+nodename = "230706_M67_a"
+
+
+# {{{ load data
+Ep = find_file(f"{h5_file}", exp_type="AG_processed_data", expno=f"{nodename}/Ep")
+# Some older h5 files save the T1p rather than the R1p. If there isn't an R1p expno then it will load the T1p integrals and convert to R1p by taking the inverse
+try:
+    R1p = find_file(f"{h5_file}", exp_type="AG_processed_data", expno=f"{nodename}/R1p")
+except:
+    T1p = find_file(f"{h5_file}", exp_type="AG_processed_data", expno=f"{nodename}/T1p")
+    R1p = T1p**-1
+# }}}
+# {{{Find the index where the return powers begin
+flip_idx = np.where(np.diff(Ep.fromaxis("power").data) < 0)[0][0] + 1
+# }}}
+with figlist_var() as fl:
+    # {{{Plot Ep
+    fl.next("Integrated Enhancement")
+    fl.plot(
+        Ep["power", :flip_idx],
+        "o",
+        label="Progressive Saturation Data",
+        capsize=6,
+        alpha=0.5,
+    )
+    fl.plot(
+        Ep["power", flip_idx:],
+        "rx",
+        label="Returning Power Check",
+        capsize=6,
+        alpha=0.5,
+    )
+    # }}}
+    # {{{ plot R1p data
+    fl.next(r"$R_{1}(p)$")
+    fl.plot(R1p, "o", label="Experimental Data")
+    # }}}
+    # {{{load in T100 dataset
+    T10_p = loadmat(
+        search_filename(
+            "T10_DI_water_230412", exp_type="AG_processed_data", unique=True
```

Comment:
please remove all references to filenames and to explicit `exp_type` within the code itself -- move up above, where you give the main filename.

I would suggest loading into dictionaries:
```
main_data = dict(
    data_dir='AG_processed_data',
    filename='whatever.h5',
    nodename='whatever',
    )
```
etc. but if you have an equally legible way of doing it, that's good, too.